Extract version to version.txt

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27149 Extract version to version.txt**

Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

pbelevich committed:
Extract version to version.txt

Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

ezyang commented on pt_v.py:
```diff
@@ -0,0 +1 @@
+__version__ = open('version.txt', 'r').read().strip()
```
What's this?

ezyang commented:
A more common pattern is to define `__version__ = ...` directly in `torch/__init__.py` and then regex it out in the `setup.py`

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]

pbelevich committed:
Update on "Extract version to version.txt"


Extract version to version.txt and add reading version logic to setup.py and fb/torch_version.py

Differential Revision: [D17689307](https://our.internmc.facebook.com/intern/diff/D17689307/)

[ghstack-poisoned]
